bfs.py

```diff
@@ -19,12 +19,37 @@ def bfs(self, source):
                 if not neighbor.visited:
                     neighbor.cost = candidate.cost + 1
                     neighbor.visited = True
+                    neighbor.parent = candidate
                     candidates.append(neighbor)
+                    print(neighbor.name, "neighbor", candidate.name)
                     print("Added", neighbor.name, "to candidates. Cost:", neighbor.cost)
                 else:
                     print(neighbor.name, "already visited! Skipping...")
             print("----------------------------------------")
         print("---------------End of BFS---------------")
+        
+    def path_finder(self, source, destination):
+        x = 1
+        print("******************** By the use of bfs, we are initiating the path search from", source.name, "to", destination.name, "and the travel cost*******************")
+        self.bfs(source)
+        traversal = []
+        traversal.append(destination.name)
```

Comment:
It might be easier to think in terms of the node itself rather than it's parent, for example:
* Checking for `while destination != None`
* Inside while, append `destination.name`
* In this way, we don't need to push the first node separately before the loop
* We don't have to track both the node and it's parent in our minds while reading/writing the code. Assign `destination = destination.parent` once, work with `destination` everywhere else.

N.B. These suggestions mentioned are totally optional, current approach is great too.